frontend/ui/translator.lua

```diff
@@ -232,11 +236,33 @@ function Translator:genSettingsMenu()
                     local lang = G_reader_settings:readSetting("translator_from_language")
                     return T(_("Translate from: %1"), self:getLanguageName(lang, ""))
                 end,
+                help_text = _("If a specific source language is manually selected, it will be used everywhere, in all your books."),
                 enabled_func = function()
                     return not G_reader_settings:nilOrTrue("translator_from_auto_detect")
+                        and not (G_reader_settings:isTrue("translator_from_doc_lang") and self:getDocumentLanguage() ~= nil)
                 end,
                 sub_item_table = genLanguagesItems("translator_from_language"),
                 keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    local __, name = self:getDocumentLanguage()
+                    return T(_("Translate from book language: %1"), name or _("N/A"))
+                end,
+                help_text = _([[
+With books that specify their main language in their metadata (most EPUBs and FB2s), enabling this option will make this language the source language (otherwise, auto-detection or the specified languages will be used).
```

Comment:
Don't go overboard with the parentheses. :-)
```suggestion
With books that specify their main language in their metadata (most EPUBs and FB2s), enabling this option will make this language the source language. Otherwise, auto-detection or the specified languages will be used.
```